server/viame_server/serializers/viame.py

```diff
@@ -189,18 +146,54 @@ def export_tracks_as_csv(file, excludeBelowThreshold, thresholds, filenames=None
     track_json = json.loads(
         b"".join(list(File().download(file, headers=False)())).decode()
     )
+    csvFile = io.StringIO()
+    writer = csv.writer(csvFile)
+    for t in track_json.values():
+        track = from_dict(Track, t, config=Config(cast=[Tuple]))
+        if (not excludeBelowThreshold) or track.exceeds_thresholds(thresholds):
+            for index, keyframe in enumerate(track.features):
+                features = [keyframe]
+
+                # If this is not the last keyframe, and interpolation is
+                # enabled for this keyframe, interpolate
+                if keyframe.interpolate and index < len(track.features) - 1:
+                    nextKeyframe = track.features[index + 1]
+                    # interpolate all features in [a,b)
+                    features = interpolate(keyframe, nextKeyframe)
+
+                for feature in features:
+                    columns = [
+                        track.trackId,
+                        "",
+                        feature.frame,
+                        *feature.bounds,
+                        track.confidencePairs[-1][1],
+                        feature.fishLength or -1,
+                    ]
```

Comment:
This one might be hard to explain but we may want a check in here to see if there are existing bounds for the keyframe and not calculate the interpolation.  If you go through and turn on interpolation for consecutive keyframes in the UI, you get doubled up keyframe bounds in the CSV file.  Once for the true bounds and then because it is only checking that interpolate flag it will create another one for the interpolated bounds.  I noticed this when I was getting double frame[1] items in the CSV.  I don't think this has an effect on the UI because it will default to the bounds if they exist.  Maybe part of the interpolate function checks and sees if there are existing bounds for both and doesn't add any.
![image](https://user-images.githubusercontent.com/61746913/91300145-ca068600-e770-11ea-9326-df4dff6a3de0.png)

Reply:
As usual, your attention to detail is astounding.  This isn't code I changed but I can fix it while I'm here.   I think I wrote it initially, but I'm not sure.